onnxscript/values.py

```diff
@@ -142,6 +142,70 @@ class OnnxClosure:
     function: Any
 
 
+def adapt_to_eager_mode(inputs):
```

Comment:
Should we hide this function? Same for the other

```suggestion
def _adapt_to_eager_mode(inputs: Union[np.ndarray, tensor.Tensor, bool, Number, list[Any], tuple[Any, ...]):
```

Reply:
For the type, the ideal would be to use a recursive type definition (https://github.com/python/mypy/issues/731) ... do you think we can use it, or is it too experimental still?

Reply:
From the comments it's one by default for mypy 0.990. since we are using 0.991 it may be worth a try